app/models/host_status/execution_status.rb

```diff
@@ -37,6 +41,47 @@ def to_label(options = {})
     end
   end
 
+  class ExecutionTaskStatusMapper
+    attr_accessor :task
+
+    def self.sql_conditions_for(status)
+      status = STATUS_NAMES.key(status) if status.is_a?(String)
+
+      case status
+        when OK
+          [ "state = 'stopped' AND (result = 'success' OR result = 'warning')" ]
```

Comment:
But warning does not mean there's some host that failed? That would end up as error I thought. I think there's not even many situation in rex in which a task can end up with warning, only I can think of is some dynflow issue after which you have to resume the task manually? If I get it wrong and warning always means that something failed and was not fixed, then I'm happy to change it to ERROR.